src/Serializer.php

```diff
@@ -272,4 +277,24 @@ protected function getIncluded()
         $include = $this->request->get($this->expandParam);
         return is_string($include) ? preg_split('/\s*,\s*/', $include, -1, PREG_SPLIT_NO_EMPTY) : [];
     }
+
+    /**
+     * Checks Identification of resource object.
+     * http://jsonapi.org/format/#document-resource-object-identification
+     * @param ResourceInterface $resource
+     */
+    protected function checkIdentification(ResourceInterface $resource)
+    {
+        if (!is_string($resource->getId())) {
```

Comment:
> And i agree that id and type MUST be converted into a string

Cool :+1: The reason I spoke up was because I'm excited about this library and am hoping to use it for a project I'm working on long-term. Really nice to see what you have here so far. 

Re flexible, I would argue that the goal of the spec is to remove flexibility on anything the spec says MUST on to make implementations more consistent and interoperable.

Do we need to worry about custom clients who might expect an integer? The nature of JSON API means we might be sending UUIDs back and the client shouldn't care ...